Source/WebKit/webpushd/PushService.mm

```diff
@@ -55,32 +61,69 @@ static void updateTopicLists(PushServiceConnection& connection, PushDatabase& da
 
 #if HAVE(APPLE_PUSH_SERVICE_URL_TOKEN_SUPPORT)
 
+static void performAfterFirstUnlock(Function<void()>&& function)
+{
+#if PLATFORM(IOS) && USE(APPLE_INTERNAL_SDK)
+    if (MKBDeviceUnlockedSinceBoot()) {
+        function();
+        return;
+    }
+
+    RELEASE_LOG(Push, "Device is locked. Delaying init until it unlocks for the first time.");
+
+    static std::once_flag once;
+    static NeverDestroyed<Vector<Function<void()>>> functions;
+
+    std::call_once(once, []() {
+        int notifyToken;
+        notify_register_dispatch(kMobileKeyBagLockStatusNotifyToken, &notifyToken, dispatch_get_main_queue(), makeBlockPtr([function = WTFMove(function)](int token) mutable {
+            if (!notify_is_valid_token(token) || !MKBDeviceUnlockedSinceBoot())
+                return;
+
+            RELEASE_LOG(Push, "Device has unlocked. Running delayed initialization.");
+
+            for (auto &function : functions.get())
+                function();
+            functions->clear();
+
+            notify_cancel(token);
+        }).get());
+    });
+
+    functions->append(WTFMove(function));
```

Comment:
On the first call to `performAfterFirstUnlock`, you changed the lambda capture list on line 77 to not WTFMove `function`, but `makeBlockPtr` still grabs it.

This is (bizarrely and problematically) still allowed because of how Obj-C blocks work.

The capture is simply not needed, because the function that's setup inside walks the `functions` Vector and doesn't care about the initial `function` argument.

Reply:
Yep, I actually submitted a new version of the PR to fix this in the same minute as this comment.